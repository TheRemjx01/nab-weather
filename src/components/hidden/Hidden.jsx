import React from 'react';

const Hidden = ({when, children}) => {
 if (when) {
  return null
 }

 return children;
};

export default Hidden;