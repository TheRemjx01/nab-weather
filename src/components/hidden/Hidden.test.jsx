import React from 'react'
import { shallow } from 'enzyme'
import Hidden from './Hidden'

const Dumb = () => <p>Dumb</p>

describe('<Hidden />', () => {
 it('should be able to prevent render if match condition', () => {
  const Wrapper = shallow(<Hidden when={true}><Dumb/></Hidden>)
  expect(Wrapper).toMatchSnapshot();
 })

 it('Should be able to render if condition not match', () => {
  const Wrapper = shallow(<Hidden when={false}><Dumb/></Hidden>)
  expect(Wrapper).toMatchSnapshot();
 })
})