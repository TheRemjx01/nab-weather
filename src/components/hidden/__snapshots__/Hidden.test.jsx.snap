// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hidden /> Should be able to render if condition not match 1`] = `ShallowWrapper {}`;

exports[`<Hidden /> should be able to prevent render if match condition 1`] = `ShallowWrapper {}`;
